packages/fast-usdc/test/fast-usdc.contract.test.ts

```diff
@@ -205,23 +220,310 @@ test('LP deposits, earns fees, withdraws', async t => {
+    );
+    await t.throwsAsync(
+      E(creatorFacet).testRepay(repayAmounts, repayPayments),
+      {
+        message: /Cannot repay. Principal .* exceeds encumberedBalance/,
+      },
+    );
+  }
+
+  {
+    const pmt = utils.pourPayment(usdc.make(50n));
+    await t.throwsAsync(
+      E(creatorFacet).testRepay(
+        // @ts-expect-error intentionally incorrect KWR
+        { USDC: usdc.make(50n) },
+        { USDC: pmt },
+      ),
+      {
+        message:
+          /Must have missing properties \["Principal","PoolFee","ContractFee"\]/,
+      },
+    );
+  }
+  {
+    const pmt = utils.pourPayment(usdc.make(50n));
+    await t.throwsAsync(
+      E(creatorFacet).testRepay(
+        // @ts-expect-error intentionally incorrect KWR
+        { Principal: usdc.make(50n) },
+        { Principal: pmt },
+      ),
+      {
+        message: /Must have missing properties \["PoolFee","ContractFee"\]/,
+      },
+    );
+  }
+  {
+    const amts = {
+      Principal: usdc.make(0n),
+      ContractFee: usdc.make(0n),
+      PoolFee: usdc.make(0n),
+    };
+    const pmts = await deeplyFulfilledObject(
+      objectMap(amts, utils.pourPayment),
+    );
+    await t.throwsAsync(E(creatorFacet).testRepay(amts, pmts), {
+      message: /arg 1: Principal: value: "\[0n\]" - Must be >= "\[1n\]"/,
+    });
+  }
+
+  {
+    t.log('repay fails when amounts do not match seat allocation');
+    const amts = {
+      Principal: usdc.make(25n),
+      ContractFee: usdc.make(1n),
+      PoolFee: usdc.make(2n),
+    };
+    const pmts = await deeplyFulfilledObject(
+      harden({
+        Principal: utils.pourPayment(usdc.make(24n)),
+        ContractFee: utils.pourPayment(usdc.make(1n)),
+        PoolFee: utils.pourPayment(usdc.make(2n)),
+      }),
+    );
+    await t.throwsAsync(E(creatorFacet).testRepay(amts, pmts), {
+      message: /Cannot repay. From seat allocation .* does not equal amounts/,
+    });
+  }
+
+  {
+    t.log('repay succeeds with no Pool or Contract Fee');
+    const amts = {
+      Principal: usdc.make(25n),
+      ContractFee: usdc.make(0n),
+      PoolFee: usdc.make(0n),
+    };
+    const pmts = await deeplyFulfilledObject(
+      objectMap(amts, utils.pourPayment),
+    );
+    const repayResult = await E(creatorFacet).testRepay(amts, pmts);
+
+    for (const r of Object.values(repayResult)) {
+      t.is(r.value, 0n, 'testRepay consumes all payments');
+    }
+  }
+
+  const amts = {
+    Principal: usdc.make(25n),
+    ContractFee: usdc.make(1n),
+    PoolFee: usdc.make(2n),
+  };
+  const pmts = await deeplyFulfilledObject(objectMap(amts, utils.pourPayment));
+  const repayResult = await E(creatorFacet).testRepay(amts, pmts);
+
+  for (const r of Object.values(repayResult)) {
+    t.is(r.value, 0n, 'testRepay consumes all payments');
+  }
+
+  await eventLoopIteration();
+  t.like(await E(subscriber).getUpdateSince(), {
+    value: {
+      encumberedBalance: {
+        value: 0n,
+      },
```

Comment:
how did the `encumberedBalance` get to zero? and if it's zero, how can alice withdraw below?

Reply:
We expect `0n` here since all borrows have been returned